src/main/java/wolox/reactortraining/services/BotService.java

```diff
@@ -16,6 +32,50 @@
     @Qualifier("bot-api")
     private WebClient botWebClient;
 
+    @Autowired
+    private TopicRepository topicRepository;
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private TwitterService twitterService;
+
+    private Logger logger = LoggerFactory.getLogger(BotService.class);
+
+    private static TopicDto createTopicDto(String description) {
+        TopicDto topicDto = new TopicDto();
+        topicDto.setDescription(description);
+        return topicDto;
+    }
+
+    private static boolean tweetHasTopics(String tweet, List<String> topics) {
+        boolean contains = false;
+
+        for (String topic : topics) {
```

Comment:
you should use operations lambda, don't use for, use a foreach

Reply:
The forEach does not allow me to return the value of contains to the upper function `tweetHasTopics` without turning the variable `contains` an Atomic reference. Should I replace that with this last alternative?